packages/multi-test/src/contracts.rs

```diff
@@ -334,10 +336,9 @@ where
         env: Env,
         info: MessageInfo,
         msg: Vec<u8>,
-    ) -> Result<Response<C>, String> {
-        let msg = from_slice(&msg).map_err(|e| e.to_string())?;
-        let res = (self.execute_fn)(deps, env, info, msg);
-        res.map_err(|e| e.to_string())
+    ) -> Result<Response<C>> {
+        let msg = from_slice(&msg)?;
+        (self.execute_fn)(deps, env, info, msg).map_err(Into::into)
```

Comment:
I guess this can be simplified further (probably in another iteration) by making `E1` and friends  simply an `anyhow::Error` type, instead of an `Into` trait.

Reply:
Possible, I am not happy on how much bounds are there. I didn't spend too much time here, but I think this particular module could use some love.

Reply:
If we are wrapping everything, we could probably use some `Box<dyn Trait>` over generics to reduce the number of bounds. Or maybe some more clever approaches. But yeah, that is another issue.